tests/unit/test_exec.py

```diff
@@ -1,9 +1,56 @@
 __copyright__ = "Copyright (c) 2020-2021 Jina AI Limited. All rights reserved."
 __license__ = "Apache-2.0"
 
-from jina.executors import BaseExecutor
+import pytest
+from jina import Document, DocumentArray
+from jinahub.encoder.transformer_tf_text_encode import TransformerTFTextEncoder
 
+target_dim = 768
 
-def test_exec():
-    ex = BaseExecutor.load_config('../../config.yml')
-    assert ex._dim == 128
+
+@pytest.fixture()
+def docs_generator():
+    return DocumentArray((Document(text='random text') for _ in range(30)))
+
+
+def test_tf_batch(docs_generator):
+    encoder = TransformerTFTextEncoder()
+    docs = docs_generator
+    encoder.encode(docs, parameters={'batch_size': 10, 'traversal_paths': ['r']})
+
+    assert len(docs.get_attributes('embedding')) == 30
+    assert docs[0].embedding.shape == (target_dim,)
+
+
+def test_traversal_path():
+    text = 'blah'
+    docs = DocumentArray([Document(id='root1', text=text)])
+    docs[0].chunks = [Document(id='chunk11', text=text),
+                      Document(id='chunk12', text=text),
+                      Document(id='chunk13', text=text)
+                      ]
+    docs[0].chunks[0].chunks = [
+        Document(id='chunk111', text=text),
+        Document(id='chunk112', text=text),
+    ]
+
+    encoder = TransformerTFTextEncoder()
+    encoder.encode(docs, parameters={'batch_size': 10, 'traversal_paths': ['c']})
+
+    for path, count in [[['r'], 0], [['c'], 3], [['cc'], 0]]:
+        assert len(docs.traverse_flat(path).get_attributes('embedding')) == count
+        if count > 0:
+            assert docs.traverse_flat(path).get_attributes('embedding')[0].shape == (target_dim,)
+
+    encoder.encode(docs, parameters={'batch_size': 10, 'traversal_paths': ['cc']})
+    for path, count in [[['r'], 0], [['c'], 3], [['cc'], 2]]:
+        assert len(docs.traverse_flat(path).get_attributes('embedding')) == count
+        if count > 0:
+            assert docs.traverse_flat(path).get_attributes('embedding')[0].shape == (target_dim,)
```

Comment:
The `traversal_paths` could be parameterized and `traversal_paths=['r']` could be added.
Similar to here : https://github.com/jina-ai/executor-image-torch-encoder/blob/65fee77edf3d7758255c0ff38c39ad4e9693ece3/tests/integration/test_flow_integration.py#L48

Reply:
how about now?

Reply:
Perfect, thanks!